log/logger.go

```diff
@@ -42,13 +30,14 @@ func InitLogger(config LogConfig) {
 		developmentCfg := zap.NewDevelopmentEncoderConfig()
 		developmentCfg.EncodeLevel = zapcore.CapitalColorLevelEncoder
 
-		consoleEncoder := zapcore.NewConsoleEncoder(developmentCfg)
-		fileEncoder := zapcore.NewJSONEncoder(productionCfg)
+		var encoder zapcore.Encoder
+		if config.IsProduction {
```

Comment:
this can be simplified:
```
encoder = zapcore.NewConsoleEncoder(developmentCfg)
if config.IsProduction {
  encoder = zapcore.NewJSONEncoder(productionCfg)
}
```

Reply:
I'd rename `IsProduction` to `UseJSONLogFormat`

Reply:
In that case, can we entirely remove the `development` config? And just keep only one configuration. And then according to the log format flag, we can switch the `Encoder`
```go
config := zap.NewProductionEncoderConfig()
var encoder zapcore.Encoder
encoder = zapcore.NewConsoleEncoder(config)
if config.IsProduction {
	encoder = zapcore.NewJSONEncoder(config)
}
```

Reply:
that makes sense, I always try to avoid naming something in an application based on environment name and keep things generic